x-pack/plugins/fleet/cypress/integration/agent.ts

```diff
@@ -0,0 +1,181 @@
+  default_api_key_id: 'abcd',
+  policy_output_permissions_hash: 'somehash',
+  updated_at: '2022-03-07T16:35:03Z',
+  last_checkin_status: 'online',
+  last_checkin: '2022-03-07T16:34:56Z',
+  policy_revision_idx: 1,
+  policy_coordinator_idx: 1,
+  policy_revision: 1,
+  status: 'online',
+  packages: [],
+};
+
+describe('View agents', () => {
+  beforeEach(() => {
+    cy.intercept('/api/fleet/agents/setup', { isReady: true });
+    cy.intercept('/api/fleet/setup', { isInitialized: true, nonFatalErrors: [] });
+    cy.intercept(/\/api\/fleet\/agent_policies(\?.*)?$/, {
+      items: [
+        {
+          id: 'policy-1',
+          name: 'Agent policy 1',
+          description: '',
+          namespace: 'default',
+          monitoring_enabled: ['logs', 'metrics'],
+          status: 'active',
+        },
+        {
+          id: 'policy-2',
+          name: 'Agent policy 2',
+          description: '',
+          namespace: 'default',
+          monitoring_enabled: ['logs', 'metrics'],
+          status: 'active',
+        },
+      ],
+    });
+    cy.intercept('/api/fleet/agent_status', {
+      results: {
+        total: 1,
+        inactive: 0,
+        online: 1,
+        error: 0,
+        offline: 0,
+        updating: 0,
+        other: 0,
+        events: 0,
+      },
+    });
+    cy.intercept(/\/api\/fleet\/agents(\?.*)?$/, {
+      items: [agent],
+      list: [agent],
+      total: 1,
+    }).as('getAgents');
+  });
+
+  describe('Agent filter suggestions', () => {
+    it('should show correct suggestions for agent.id', () => {
+      cy.intercept(/\/api\/index_patterns*/, {
+        fields: [
+          {
+            name: 'agent.id',
+            type: 'string',
+            esTypes: ['keyword'],
+            searchable: true,
+            aggregatable: true,
+            readFromDocValues: true,
+            metadata_field: false,
+          },
+        ],
+      });
+      cy.intercept('/api/kibana/suggestions/values/.fleet-agents', [
+        'suggestion1',
+        'suggestion2',
+      ]).as('getSuggestions');
+
+      cy.visit('/app/fleet/agents');
+      cy.get('[data-test-subj="queryInput"]').clear().type('agent.id: ');
+
+      cy.wait('@getSuggestions').then(({ request }) => {
+        expect(request.body.field).to.equal('agent.id');
+      });
+
+      cy.get('.kbnTypeahead').within(() => {
+        cy.contains('suggestion1');
+        cy.contains('suggestion2');
+      });
+    });
+  });
+
+  describe('Upgrade available filter', () => {
+    it('should call API with correct query param on click ', () => {
+      cy.visit('/app/fleet/agents');
+      cy.contains('agent-1');
+
+      cy.get('[data-test-subj="agentList.showUpgradeable"]').click();
+      // getAgents can be called many times on load, wait() was grabbing the wrong call
+      // .all returns all requests performed so far, allowing us to reliably grab the most recent call
+      cy.get('@getAgents.all').then((requests) => {
```

Comment:
maybe you can move out this logic to a utility `getLatestRequest` that takes the request name as parameter